plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -99,11 +81,24 @@ function auto_sizes_filter_image_tag( $content, array $parsed_block, WP_Block $b
 		 * @param string $size  The image size data.
 		 */
 		$filter = static function ( $sizes, $size ) use ( $block ) {
-			$id        = $block->attributes['id'] ?? 0;
-			$alignment = $block->attributes['align'] ?? '';
-			$width     = $block->attributes['width'] ?? '';
 
-			return auto_sizes_calculate_better_sizes( (int) $id, (string) $size, (string) $alignment, (string) $width );
+			$id            = $block->attributes['id'] ?? 0;
+			$alignment     = $block->attributes['align'] ?? '';
+			$width         = $block->attributes['width'] ?? '';
+			$max_alignment = $block->context['max_alignment'];
+
+			/*
+			 * Update width for cover block.
+			 * See https://github.com/WordPress/gutenberg/blob/938720602082dc50a1746bd2e33faa3d3a6096d4/packages/block-library/src/cover/style.scss#L82-L87.
+			 */
+			if ( 'core/cover' === $block->name && in_array( $alignment, array( 'left', 'right' ), true ) ) {
+				$size = array( 420, 420 );
+			}
+
+			$better_sizes = auto_sizes_calculate_better_sizes( (int) $id, $size, (string) $alignment, (string) $width, (string) $max_alignment );
```

Comment:
Question: Why is `$width` a `string`? Should above it instead do:

```php
$width = $block->attributes['width'] ?? 0;
```

And then here it can do:

```suggestion
			$better_sizes = auto_sizes_calculate_better_sizes( (int) $id, $size, (string) $alignment, (int) $width, (string) $max_alignment );
```

And then some code in `auto_sizes_calculate_better_sizes()` can be removed below.

Reply:
This value comes from the image block's `width` attribute, which is a string ([see](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/image/block.json#L73-L75)). However, it looks like only px values are supported, so we could consider converting these values. I'll git it a try.

Reply:
Looks like this and the suggested changes all still work as expected. Applied in aecb3de9